cli/cli_app.go

```diff
@@ -182,7 +188,17 @@ func runTerragrunt(terragruntOptions *options.TerragruntOptions) error {
 	}
 
 	if conf.Terraform != nil && conf.Terraform.ExtraArgs != nil && len(conf.Terraform.ExtraArgs) > 0 {
-		terragruntOptions.TerraformCliArgs = append(terragruntOptions.TerraformCliArgs, filterTerraformExtraArgs(terragruntOptions, conf)...)
+		commandLength := 1
+		if util.ListContainsElement(TERRAFORM_COMMANDS_WITH_SUBCOMMAND, terragruntOptions.TerraformCliArgs[0]) {
+			commandLength = 2
+		}
+		var args []string
+		args = append(args, terragruntOptions.TerraformCliArgs[:commandLength]...)
+
+		// Options must be inserted after command but before the other args
+		// command is either 1 word or 2 words
+		args = append(args, filterTerraformExtraArgs(terragruntOptions, conf)...)
+		terragruntOptions.TerraformCliArgs = append(args, terragruntOptions.TerraformCliArgs[commandLength:]...)
```

Comment:
It would be more readable to use a consistent structure for building up the list of `args`:

```go
var args []string

// Options must be inserted after command but before the other args
// command is either 1 word or 2 words
args = append(args, terragruntOptions.TerraformCliArgs[:commandLength]...)
args = append(args, filterTerraformExtraArgs(terragruntOptions, conf)...)
args = append(args, terragruntOptions.TerraformCliArgs[commandLength:]...)

terragruntOptions.TerraformCliArgs = args
```